plugins/hypervisors/simulator/src/com/cloud/agent/manager/MockVmManagerImpl.java

```diff
@@ -449,12 +450,34 @@ public Answer CleanupNetworkRules(final CleanupNetworkRulesCmd cmd, final Simula
 
     @Override
     public Answer scaleVm(final ScaleVmCommand cmd) {
-        return null;  //To change body of implemented methods use File | Settings | File Templates.
+        TransactionLegacy txn = TransactionLegacy.open(TransactionLegacy.SIMULATOR_DB);
+        try {
+            txn.start();
+            final String vmName = cmd.getVmName();
+            final MockVMVO vm = _mockVmDao.findByVmName(vmName);
+            if (vm == null) {
+                return new ScaleVmAnswer(cmd, false, "Can't find VM " + vmName);
+            }
+            vm.setCpu(cmd.getCpus() * cmd.getMaxSpeed());
+            vm.setMemory(cmd.getMaxRam());
+            _mockVmDao.update(vm.getId(), vm);
+            s_logger.debug("Scaled up VM " + vmName);
+            txn.commit();
+            return new ScaleVmAnswer(cmd, true, null);
+        } catch (final Exception ex) {
+            txn.rollback();
+            throw new CloudRuntimeException("Unable to scale up VM", ex);
+        } finally {
+            txn.close();
+            txn = TransactionLegacy.open(TransactionLegacy.CLOUD_DB);
```

Comment:
Just out of curiosity: why do you open and immediately close a transaction in the `finally` block of another transaction?